src/app/beer_garden/db/mongo/models.py

```diff
@@ -457,6 +457,20 @@ def clean(self):
                 f"Can not save Request {self}: Invalid output type '{self.output_type}'"
             )
 
+        # Deal with has_parent
+        if self.has_parent is None:
+            self.has_parent = bool(self.parent)
+        elif self.has_parent is False and self.parent:
```

Comment:
You could combine the two else blocks here no?

```python
elif self.has_parent != bool(self.parent):
    raise ModelValidationError(
        f"Cannot save Request {self}: parent value of {self.parent} is not consistent with "
        f"has_parent value of {self.has_parent}"
    )
```

Something like that.

Reply:
Sure, that's fine.